src/iceberg/manifest/manifest_reader.cc

```diff
@@ -19,61 +19,870 @@
 
 #include "iceberg/manifest/manifest_reader.h"
 
+#include <algorithm>
+#include <memory>
+#include <unordered_set>
+
+#include <nanoarrow/nanoarrow.h>
+
+#include "iceberg/arrow/nanoarrow_status_internal.h"
+#include "iceberg/arrow_c_data_guard_internal.h"
+#include "iceberg/expression/expression.h"
+#include "iceberg/expression/projections.h"
+#include "iceberg/file_format.h"
 #include "iceberg/manifest/manifest_entry.h"
 #include "iceberg/manifest/manifest_list.h"
 #include "iceberg/manifest/manifest_reader_internal.h"
+#include "iceberg/metadata_columns.h"
+#include "iceberg/partition_spec.h"
 #include "iceberg/schema.h"
-#include "iceberg/schema_internal.h"
+#include "iceberg/schema_field.h"
 #include "iceberg/type.h"
+#include "iceberg/util/checked_cast.h"
 #include "iceberg/util/macros.h"
 
 namespace iceberg {
 
+namespace {
+
+#define PARSE_PRIMITIVE_FIELD(item, array_view, type)                                   \
```

Comment:
Github is not clever enough to figure out changes in this file. Most parts come from combining `manifest_reader.cc` and `manifest_reader_internal.cc`.

Reply:
Can we change these macros to template functions?

Reply:
I agree with you. But I prefer to do it in a separate PR.